### PR TITLE
Add support for terraform language

### PR DESCRIPTION
**Pull request by danlamanna**
This is based on #69, better diff: [pokey/reinstate-local-web-tree-sitter-build..danlamanna/add-terraform-language.](https://github.com/cursorless-dev/vscode-parse-tree/compare/pokey/reinstate-local-web-tree-sitter-build...danlamanna:vscode-parse-tree:add-terraform-language)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,6 +37,10 @@ parsers/tree-sitter-c-sharp.wasm: node_modules/tree-sitter-c-sharp/package.json
 	npx tree-sitter build-wasm $(dir $^)
 	mv tree-sitter-c_sharp.wasm $@
 
+parsers/tree-sitter-terraform.wasm: node_modules/tree-sitter-hcl/package.json
+	mkdir -p $(dir $@)
+	npx tree-sitter build-wasm $(dir $^)
+	mv tree-sitter-hcl.wasm $@
```
Comment (pokey) [inline Makefile:43]: Why not just remove this block and let it be called `tree-sitter-hcl`?

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # For generating .wasm files for parsers
 # See https://www.npmjs.com/package/web-tree-sitter
-LANGUAGES = agda bash c c-sharp clojure cpp css elm elixir go haskell html java javascript json julia kotlin latex markdown perl php python query ruby rust scala scss sparql talon tsx typescript yaml
+LANGUAGES = agda bash c c-sharp clojure cpp css elm elixir go haskell html java javascript json julia kotlin latex markdown perl php python query ruby rust scala scss sparql talon terraform tsx typescript yaml
```
Comment (pokey) [inline Makefile:3]: Per my other comment, I'd be tempted to change this to `hcl`, unless I'm missing something

File: src/extension.ts [hunk 3]
```diff
@@ -44,6 +44,7 @@ const languages: {
   sparql: { module: "tree-sitter-sparql" },
   starlark: { module: "tree-sitter-python" },
   talon: { module: "tree-sitter-talon" },
+  terraform: { module: "tree-sitter-hcl" },
```
Comment (pokey) [inline src/extension.ts:47]: I'm actually a bit surprised this works, given your Makefile. Shouldn't this be `tree-sitter-hcl`? That being said, if you implement my other suggestions, you should leave this line alone. Just a bit surprised it works as is

**Review by pokey — COMMENTED** [review]
Mostly looks good; left a few comments / questions

File: package.json [hunk 4]
```diff
@@ -79,25 +80,28 @@
...
     "tree-sitter-cli": "^0.20.1",
     "tree-sitter-clojure": "github:sogaiu/tree-sitter-clojure#master",
     "tree-sitter-cpp": "^0.19.0",
-    "@elm-tooling/tree-sitter-elm": "github:elm-tooling/tree-sitter-elm#0aecfbf69f0c9bcdfada0f6f2c5398235c23949f",
-    "tree-sitter-go": "github:tree-sitter/tree-sitter-go#bbaa67a180cfe0c943e50c55130918be8efb20bd",
+    "tree-sitter-css": "^0.19.0",
     "tree-sitter-elixir": "github:elixir-lang/tree-sitter-elixir#b20eaa75565243c50be5e35e253d8beb58f45d56",
+    "tree-sitter-go": "github:tree-sitter/tree-sitter-go#bbaa67a180cfe0c943e50c55130918be8efb20bd",
     "tree-sitter-haskell": "github:tree-sitter/tree-sitter-haskell#d6ccd2d9c40bdec29fee0027ef04fe5ff1ae4ceb",
+    "tree-sitter-hcl": "https://github.com/MichaHoffmann/tree-sitter-hcl",
```
Comment (pokey) [inline package.json:104]: please pint to a specific sha. tree-sitter grammars regularly make breaking changes 😅

**Comment by pokey** [comment]
whoops looks like this got deleted / closed when I merged #69

**Comment by danlamanna** [comment]
> whoops looks like this got deleted / closed when I merged #69. Are you able to recreate the branch? There should be a button on this PR, but I don't seem to have it on my side 🤔

Nope, PRs have to be recreated unfortunately :(.

**Pull request closed without merging**